src/main/java/com/example/petbuddybackend/service/care/state/CareStateMachine.java

```diff
@@ -53,6 +61,25 @@ public void transitionToEditCare(Care care) {
         care.setClientStatus(CareStatus.PENDING);
     }
 
+    @Transactional
+    public void cancelCaresIfStatePermitsAndSave(String firstUsername, String secondUsername) {
+        cancelCaresOfClientAndCaretakerAndSave(firstUsername, secondUsername);
+        cancelCaresOfClientAndCaretakerAndSave(secondUsername, firstUsername);
+    }
+
+    @Transactional
+    public int outdateCaresIfStatePermitsAndSave() {
+        return careRepository.outdateCaresBetweenClientAndCaretaker(globalStatusesThatCanBeOutdated);
```

Comment:
em, wszystko fajnie, tylko, ze to o polnocy zrobi outdate na wszystkich opiekach z podanymi stausami xD
Musisz jeszcze pyknac warunek, najlepiej juz w tym query, zeby zrobilo outdate dla opiek z podanymi statusami AND jesli startDate minął własnie :)

Reply:
Tak, zapomniałem o tym warunku, dzięki za zwrócenie uwagi. Za bardzo się pośpieszyłem xD

Reply:
Z drugiej strony testy przechodziły :+1: 
(sam je pisałem)